server/map_tree_updater.go

```diff
@@ -110,44 +109,36 @@ func (t *mapTreeUpdater) update(ctx context.Context, tx storage.MapTreeTX, nodes
 }
 
 type txAccessor struct {
-	// hashes is a cache of node hashes that Get returns directly instead of
-	// calling GetMerkleNodes. Used only if preload is true.
-	hashes  map[tree.NodeID2][]byte
-	preload bool
-
-	tx  storage.MapTreeTX
-	rev int64
+	mtu *mapTreeUpdater
+	// read is a cache of tree tiles that Get returns directly instead of calling
+	// GetTiles. It is initialized by the first Get call if not specified.
+	read *smt.TileSet
+	tx   storage.MapTreeTX
+	rev  int64
 }
 
-func (t txAccessor) Get(ctx context.Context, ids []tree.NodeID2) (map[tree.NodeID2][]byte, error) {
+// TODO(pavelkalinnikov): Get should take a list of tile IDs.
+func (t *txAccessor) Get(ctx context.Context, ids []tree.NodeID2) (map[tree.NodeID2][]byte, error) {
 	// TODO(pavelkalinnikov): Factor out preload into another accessor.
-	if t.preload {
-		return t.hashes, nil
-	}
-
-	// TODO(pavelkalinnikov): Pass NodeID2 directly to storage.
-	convIDs := make([]tree.NodeID, 0, len(ids))
-	for _, id := range ids {
-		convIDs = append(convIDs, tree.NewNodeIDFromID2(id))
-	}
-	nodes, err := t.tx.GetMerkleNodes(ctx, t.rev-1, convIDs)
```

Comment:
Am I correct in understanding that `GetMerkleNodes` has no callers now?

Reply:
Could you mark that API as deprecated?

Reply:
It is still used by logs (sequencer and proofs), so it's not deprecated yet. We will delete this API in `v2` (more specifically, it will be moved to a log-specific storage API).